pkg/daemon/handler.go

```diff
@@ -57,10 +58,13 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var m messageBody
-	decoder := json.NewDecoder(r.Body)
-	err = decoder.Decode(&m)
+	b, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
```

Comment:
Just `json.NewDecoder(b).Decode(&m)` should let you do this as a one-liner (I think this is my bad for saying "Unmarshal" in the previous comment without thinking, I was mostly meaning we could make it a one-liner :))